File: Sakila_Net/Sakila_Net/VB_SakilaLib/Model/Language.vb
Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace Model
public class ClLanguage

   private languageId as Integer
   private name as String
   private lastUpdate as Date


   public Property DataLanguageId() as Integer
        Get
            Return languageId
        End Get
        Set(ByVal Value as Integer)
            languageId = Value
        End Set        
   End Property   
   public Property DataName() as String
        Get
            Return name
        End Get
        Set(ByVal Value as String)
            name = Value
        End Set        
   End Property   
   public Property DataLastUpdate() as Date
        Get
            Return lastUpdate
        End Get
        Set(ByVal Value as Date)
            lastUpdate = Value
        End Set        
   End Property   

   Public Sub New (languageId    as Integer,name    as String,lastUpdate    as Date)
		   dataLanguageId = languageId
		   dataName = name
		   dataLastUpdate = lastUpdate
   End Sub
   Public Sub New (languageId    as Integer)
    dataLanguageId = languageId
   End Sub


   public Overrides Function  ToString() As String
        '// Registro JSon
		return "{" _ 
		    & "," & chr(34) & "languageId" & chr(34) & ":" & chr(34) & dataLanguageId & chr(34) _
		    & "," & chr(34) & "name" & chr(34) & ":" & chr(34) & dataName & chr(34) _
		    & "," & chr(34) & "lastUpdate" & chr(34) & ":" & chr(34) & dataLastUpdate & chr(34) _
		 & "}"
   End Function   
   public Sub imprimir()
        Console.WriteLine(ToString())
   End Sub
   public Function jsonSerialize()
           return "get_object_vars($this);"
   End Function   
End Class
         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  

File: Sakila_Php/Entities/Language.php
<?PHP
class Language implements JsonSerializable{

   private $languageId;
   private $name;
   private $lastUpdate;


   public function getLanguageId(){
        return $this->languageId;
   }
   public function setLanguageId($languageId){
        $this->languageId = $languageId;
   }
   
   public function getName(){
        return $this->name;
   }
   public function setName($name){
        $this->name = $name;
   }
   
   public function getLastUpdate(){
        return $this->lastUpdate;
   }
   public function setLastUpdate($lastUpdate){
        $this->lastUpdate = $lastUpdate;
   }
   

   public function Language($languageId,$name='',$lastUpdate=''){
		   $this->setLanguageId($languageId);
		   $this->setName($name);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __constructXX($languageId,$name='',$lastUpdate='') {
		   $this->setLanguageId($languageId);
		   $this->setName($name);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __toString(){
        // Registro JSon
		return "{" 
		    . "," . chr(34) . "languageId" . chr(34) . ":" . chr(34) . $this->getLanguageId() . chr(34) 
		    . "," . chr(34) . "name" . chr(34) . ":" . chr(34) . $this->getName() . chr(34) 
		    . "," . chr(34) . "lastUpdate" . chr(34) . ":" . chr(34) . $this->getLastUpdate() . chr(34) 
		 . "}";
   }   
   public function imprimir()
   {
        echo $this->__toString();
   }
   public function jsonSerialize()
   {
        return get_object_vars($this);
   }   
}
// Para realizar pruebas
//$actor = new Language();
//echo "Imprimiendo ";$actor->imprimir();
//echo "ToString : " . $actor;
?>
